src/ranged.cpp

```diff
@@ -3534,63 +3519,48 @@ void target_ui::draw_controls_list( int text_y )
     std::vector<line> lines;
 
     // Compile full list
-    if( shifting_view ) {
-        lines.push_back( {8, colored( col_move, _( "Shift view with directional keys" ) )} );
-    } else {
-        lines.push_back( {8, colored( col_move, _( "Move cursor with directional keys" ) )} );
-    }
+    std::string movement_text = shifting_view ? _( "Shift view" ) : _( "Move cursor" );
+    movement_text = colorize( movement_text, input_context::get_hint_color( move_hint_state ) );
+    lines.push_back( {9, string_format( "%s %s", ctxt.get_hint_directions( move_hint_state ), movement_text )} );
     if( is_mouse_enabled() ) {
-        std::string move = _( "Mouse: LMB: Target, Wheel: Cycle," );
-        std::string fire = _( "RMB: Fire" );
-        lines.push_back( {7, colored( col_move, move ) + " " + colored( col_fire, fire )} );
+        std::string move = string_format( _( "%s %s" ), ctxt.get_hint_mouse( "SELECT", _( "Target" ),
+                                          move_hint_state ), ctxt.get_hint_mouse( "SCROLL_UP", _( "Cycle" ), move_hint_state ) );
+        std::string fire = ctxt.get_hint_mouse( "FIRE", fire_hint_state );
+        lines.push_back( {8, string_format( "%s %s", move, fire )} );
```

Comment:
These hints were displaying mouse inputs specifically. I've retained that by limiting the hints in this way because the keys to move/fire with the keyboard are displayed further down. What would like me to do here?

Reply:
Previously I implemented selecting with the mouse cursor and a keyboard key by binding a non-mouse key to the "SELECT" action, which is different from using the keyboard hotkey to fire at the currently selected target. Could you try rebinding a keyboard hotkey to the "SELECT" action and see the hints make sense in that case?